src/client/api/pythonApi.ts

```diff
@@ -291,16 +325,17 @@ export class InterpreterService implements IInterpreterService {
     ) {}
 
     public get onDidChangeInterpreter(): Event<void> {
-        if (this.extensionChecker.isPythonExtensionInstalled && !this.eventHandlerAdded) {
-            this.apiProvider
-                .getApi()
-                .then((api) => {
-                    if (!this.eventHandlerAdded) {
-                        this.eventHandlerAdded = true;
-                        api.onDidChangeInterpreter(() => this.didChangeInterpreter.fire(), this, this.disposables);
-                    }
-                })
-                .catch(noop);
+        if (this.extensionChecker.isPythonExtensionInstalled) {
+            if (this.extensionChecker.isPythonExtensionActive && this.eventHandlerAdded) {
```

Comment:
Why is the check for this.eventHandlerAdded here? If eventHandlerAdded === true then hookup is a noop.

Reply:
Fixed